### PR TITLE
[New] Payment Method Add (POST /account/payment-methods)

### PR DESCRIPTION
**Pull request by bbiggerr**
* Deprecate Credit Card Add/Edit (POST /account/credit-card)

### CODE REVIEW

**Review by nbokoski — APPROVED** [review]

File: openapi.yaml [hunk 1]
```diff
@@ -551,15 +551,9 @@ paths:
...
+        **DEPRECATED**. Please use Payment Method Add ([POST /account/payment-methods](/docs/api/account/#payment-method-add)).
 
-        Only one credit card can be associated with your Account, so using this
-        endpoint will overwrite your currently active card information with the
-        new credit card.
-
-        To use this endpoint, you must have a valid `zip` entered for your Account.
-        Use the Account Update ([PUT /account](/docs/api/account/#account-update))
-        endpoint to enter a new zip code.
+        Adds a credit card Payment Method to your account and sets it as the default method.
```
Comment (yoshizzle) [inline openapi.yaml:556]: In this comment, it is stated "and sets it as the default method"

However, in another edit below, it states "with the option to set it as the default method."

This should be clarified to the correct statement in whichever place is incorrect.

File: openapi.yaml [hunk 2]
```diff
@@ -1732,6 +1679,82 @@ paths:
...
             linode-cli account payment-methods-list
+    post:
+      servers:
+        - url: https://api.linode.com/v4
+      x-linode-grant: read_write
+      tags:
+        - Account
+      summary: Payment Method Add
+      description: |
+        Adds a Payment Method to your Account with the option to set it as the default method.
```
Comment (yoshizzle) [inline openapi.yaml:1690]: Here is the other line from the comment above

File: openapi.yaml [hunk 3]
```diff
@@ -16006,14 +16032,24 @@ components:
...
             must result in a month/year combination of the current month or in
             the future. An expiration date set in the past is invalid.
           example: 2020
+        cvv:
+          type: string
+          minLength: 3
+          maxLength: 4
+          format: digits
+          description: >
+            CVV (Card Verification Value) of the credit card, typically found on the back of the card. ****
```
Comment (yoshizzle) [inline openapi.yaml:16051]: What are the four asterisks here for?

**Review by yoshizzle — CHANGES_REQUESTED** [review]

File: openapi.yaml (hunk 1 above)
Comment (bbiggerr) [reply openapi.yaml:556]: This describes the behavior of the deprecated `PUT /account/credit-card` endpoint, which takes in a different request body that doesn't include an `is_default` field, so it currently sets the payment method as the default automatically in the backend.

File: openapi.yaml (hunk 3 above)
Comment (bbiggerr) [reply openapi.yaml:16051]: A placeholder that I forgot to remove! Removing now...

File: openapi.yaml [hunk 4]
```diff
@@ -551,15 +551,9 @@ paths:
...
-        Adds/edit credit card information to your Account.
+        **DEPRECATED**. Please use Payment Method Add ([POST /account/payment-methods](/docs/api/account/#payment-method-add)).
 
-        Only one credit card can be associated with your Account, so using this
-        endpoint will overwrite your currently active card information with the
-        new credit card.
-
-        To use this endpoint, you must have a valid `zip` entered for your Account.
-        Use the Account Update ([PUT /account](/docs/api/account/#account-update))
-        endpoint to enter a new zip code.
```
Comment (alsaihn) [inline openapi.yaml:562]: This bit is still valid and this endpoint will still report an error if the listed zip is not valid. Should we keep it?

**Review by yoshizzle — APPROVED** [review]

**Review by alsaihn — APPROVED** [review]

File: openapi.yaml (hunk 4 above)
Comment (bbiggerr) [reply openapi.yaml:562]: I reworded this a bit in the Payment Method Add description to be a bit broader. Agreed that this bit is still valid -- updating that description to be specific about requiring a valid `zip` as well.

**Pull request merged**